src/python/WMComponent/RucioInjector/RucioInjectorPoller.py

```diff
@@ -370,15 +370,61 @@ def deleteBlocks(self):
         """
         _deleteBlocks_
         Find deletable blocks, then decide if to delete based on:
-        Is there an active subscription for dataset or block ?
-          If yes => set deleted=2
-          If no => next check
         Has transfer to all destinations finished ?
-          If yes => request block deletion, approve request, set deleted=1
+          If yes => Delete rules associated with the block, set deleted=1
           If no => do nothing (check again next cycle)
         """
-        # FIXME: figure out the proper logic for rule block deletion
-        logging.info("Starting deleteBlocks methods --> IMPLEMENT-ME!!!")
+        #Get list of blocks that can be deleted
+        blockDict = self.findDeletableBlocks.execute(transaction=False)
+
+        if not blockDict:
+            logging.info("No blocks were found")
+            return
+
+        deletableBlocks = []
+        containerDict = {}
+		#Populate containerDict, assigning each block to its correspondant container
+        for blockName in blockDict:
+            containerList = self.rucio.listParentDIDs(blockName, scope=self.scope)
+            for cont in containerList:
+                if cont['name'] not in containerDict:
+                    containerDict[cont['name']] = {'blocks':[], 'rse':[], 'copies':0}
+                containerDict[cont['name']]['blocks'].append(blockName)
+
+        for contName in containerDict:
+            cont = containerDict[contName]
+            rules = self.rucio.listDataRules(contName, self.scope)
+            #Retrieve requested copies and RSEs
+            for rule in rules:
+                if rule['account'] == self.rucioAcct:
+                    cont['copies'] += rule['copies']
+                    cont['rse'].extend(self.rucio.evaluateRSEExpression(rule['rse_expression'], useCache=True, returnTape=True))
+
+            #Get RSE in which the block is available
+            availableRSEs = self.rucio.getReplicaInfoForBlocks(block=cont['blocks'])
+
+            for rses in availableRSEs:
```

Comment:
How about giving a more meanigful name to this `rses` variable, like `blockRSEs`?

Reply:
I changed the variable name to the one you suggested